tests/test_openlibrary.py

```diff
@@ -17,43 +22,67 @@
 class TestOpenLibrary(unittest.TestCase):
 
     def setUp(self):
-        ol_config = Config().get_config()
-        ol_creds = ol_config.get('s3')
-        test = not all([ol_creds.access, ol_creds.secret])
-        self.ol = OpenLibrary(credentials=ol_creds, test=test)
+        self.ol = OpenLibrary(test=True)
 
-    def test_get_olid_by_isbn(self):
+    @patch('requests.Session.get')
+    def test_get_olid_by_isbn(self, mock_get):
+        isbn_bibkeys = { 'ISBN:0374202915': { 'info_url': 'https://openlibrary.org/books/OL23575801M/Marie_LaVeau' } }
+        mock_get.return_value.json.return_value = isbn_bibkeys
         olid = self.ol.Edition.get_olid_by_isbn(u'0374202915')
         expected_olid = u'OL23575801M'
         self.assertTrue(olid == expected_olid,
                         "Expected olid %s, got %s" % (expected_olid, olid))
 
-    def test_get_work_by_metadata(self):
+    @patch('requests.Session.get')
+    def test_get_work_by_metadata(self, mock_get):
+        doc = {
+            "key":    u"/works/OL2514747W",
+            "title":  u"The Autobiography of Benjamin Franklin",
+        }
+        search_results = { 'start': 0, 'num_found': 1, 'docs': [doc] }
         title = u"The Autobiography of Benjamin Franklin"
+        mock_get.return_value.json.return_value = search_results
         book = self.ol.Work.search(title=title)
+        mock_get.assert_called_with("%s/search.json?title=%s" % (self.ol.base_url, title))
```

Comment:
I think these `assert_called_with` assertions are really useful; could we add them to the other tests as well? (`test_get_olid_by_isbn` and `test_create_book` seem to be the ones lacking them)

Reply:
Good idea, I'll add those too, thanks!